src/coreclr/vm/threads.cpp

```diff
@@ -4931,7 +4931,7 @@ void ThreadStore::AddThread(Thread *newThread)
 {
     CONTRACTL {
         NOTHROW;
-        if (GetThreadNULLOk()) {GC_TRIGGERS;} else {DISABLED(GC_NOTRIGGER);}
+        if (GetThreadNULLOk()) {DISABLED(GC_NOTRIGGER);} else {GC_TRIGGERS;}
```

Comment:
The only interesting thing that this method does is taking the thread store lock (`ThreadSuspend::LockThreadStore`). 

`ThreadSuspend::LockThreadStore` contract is https://github.com/dotnet/runtime/blob/80fb00f580f5b2353ff3a8aa78c5b5fd3f275a34/src/coreclr/vm/threadsuspend.cpp#L1857-L1858 . This contract should be same or superset of the `ThreadSuspend::LockThreadStore` contract.

I am wondering if `AddThread` and similar methods that take thread store lock should just be marked with unconditional GC_TRIGGERS to keep things simple.